bindings/rascal/models/krr.py

```diff
@@ -0,0 +1,203 @@
+        return dict()
+
+def train_gap_model(kernel, managers, KNMp, X_pseudo, y_train, self_contributions, f_train=None, lambdas=None, jitter=1e-8):
+    """
+    Defines the procedure to train a GAP model [1]:
+    .. math::
+        Y(X) = \sum_{(i,a)\in X} y_a(X_i^a),
+    where :math:`Y(X)` is the predicted property function associated with the
+    atomic structure :math:`X$, :math:`i` and :math:`a` are the index and
+    species of the atoms in structure :math:`X` and :math:`y_a(X_i^a)` is the
+    atom centered model that depends on the central atomic species.
+    The individual predictions are given by:
+    .. math::
+        y_a(X_i) = \sum_m \alpha_m k(T_m^a, X_i^a),
+    where :math:`k(\cdot,\cdot)` is a kernel function, :math:`\alpha_m` are the
+    weights of the model and :math:`T_m` are the M pseudo points used to train
+    the model.
+
+    The training is given by:
+    .. math::
+        \bm{\alpha} =  K^{-1} \bm{Y},
+    where :math:`K` is given by:
+    .. math::
+        K = K_{MM} + K_{MN} \Lambda^{-2} K_{NM},
+    :math:`\bm{Y}=K_{MN} \Lambda^{-2} \bm{y}$, :math:`\bm{y}` the training
+    targets and :math:`\Lambda` the regularization matrix.
+    The regularization matrix is chossen to be diagonal:
+    .. math::
+        \Lambda^{-1}_{nn} = \delta_{nn} * lambdas[0] / \sigma_{\bm{y}} * np.sqrt(Natoms)
+    for the targets and
+    .. math::
+        \Lambda^{-1}_{nn} = \delta_{nn} * lambdas[1] / \sigma_{\bm{y}},
+    for the derivatives of the targets w.r.t the atomic positions and
+    :math:`\sigma_{\bm{y}}` is the standard deviation of the target property
+    (not derivatives).
+
+    Parameters
+    ----------
+    kernel : Kernel
+        SparseKernel to compute KMM and initialize the model. It was used to
+        build KNMp.
+    managers : AtomsList
+        Training structures with features (and gradients)
+    KNMp : np.array
+        kernel matrix to use in the training, typically computed with:
+        KNM = kernel(managers, X_pseudo)
+        KNM_down = kernel(managers, X_pseudo, grad=(True, False))
+        KNM = np.vstack([KNM, KNM_down])
+        when training with derivatives.
+    X_pseudo : PseudoPoints
+        reference samples to use in the model
+    y_train : np.array
+        reference property
+    self_contributions : dictionary
+        map atomic number to the property baseline, e.g. isolated atoms
+        energies when the model has been trained on total energies.
+    f_train : np.array, optional
+        minus derivatives of y_train w.r.t to the atomic motion, e.g.
+        interatomic forces, by default None
+    lambdas : list/tuple, optional
+        regularisation parameter for the training, i.e. lambdas[0] -> property
+        and lambdas[1] -> gradients of the property, by default None
+    jitter : double, optional
+        small jitter for the numerical stability of solving the linear system,
+        by default 1e-8
+
+    Returns
+    -------
+    KRR
```

Comment:
it's the type of the returned object. I think that is expected in the the numpy docstring format. If it where to return a list it would be written list there.

Reply:
Ok, I see the issue.